### PR TITLE
feat(core,phrases): throw database error with meaningful message

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Throw `RequestError` about `DatabaseError` with meaningful message

- e.g. `value too long for type character varying(128)`

## Testing

- [x] UTs

image

- Locally tested.

image

### CODE REVIEW

File: packages/core/src/middleware/koa-slonik-error-handler.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 import { SchemaLike } from '@logto/schemas';
 import { Middleware } from 'koa';
+import { DatabaseError } from 'pg';
```
Comment (gao-sun) [inline packages/core/src/middleware/koa-slonik-error-handler.ts:24]: can we just import type? it'll make sense if we don't have `pg` in our deps

Comment (gao-sun) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:24]: after re-consideration i think we should not show db error in API? @simeng-li @wangsijie any ideas?

Comment (IceHe) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:24]: `DatabaseError` is a `class` not type. We cannot just import its type to deal with `DatabaseError`.

Comment (wangsijie) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:24]: It's OK to show db error in development mode, and only show 500 without details in production?

**Comment by IceHe** [comment]
Closed. We will check and throw errors using the `guard` and `createdGuard` generated by `schemas`.

**Pull request closed without merging**